tmt/steps/provision/mrack.py

```diff
@@ -236,6 +271,29 @@ def _transform_unsupported(
     return MrackHWOrGroup()
 
 
+def _translate_constraint_by_config(
+    constraint: tmt.hardware.Constraint[Any],
+    hardware_config: list[Translate],
+    logger: tmt.log.Logger,
+) -> Union[dict[str, Any], MrackBaseHWElement]:
+    name, _, child_name = constraint.expand_name()
+    config_str = None
+    for config in hardware_config:
+        if (name == config.name and child_name == config.child_name) or (
+            name == config.name and not child_name
+        ):
+            config_str = config.element
+            break
+        continue
+
+    beaker_operator, actual_value, negate = operator_to_beaker_op(
+        constraint.operator, constraint.value
+    )
+    if config_str:
+        return yaml_to_dict(config_str % (beaker_operator, actual_value))
```

Comment:
Nope, no `%`, no `format()`: tmt should stick to one templating option, this should be `render_template(config_str, ...)`.